src/Mvc/Mvc.Core/src/ApplicationModels/ActionAttributeRouteModel.cs

```diff
@@ -131,9 +131,13 @@ private static void AddEndpointMetadata(SelectorModel selector, IList<object> me
         {
             if (metadata != null)
             {
-                for (var i = 0; i < metadata.Count; i++)
+                for (var i = metadata.Count - 1; i >= 0; i--)
```

Comment:
This was messing up the `GetMetadata` call in Auth. If you had `AuthAttribute` on a controller and action, we were currently returning the value from the controller.

Reply:
ugh.

Reply:
I don't actually get this. So we have a list of metadata, and you're reversing it... this seems like the wrong fix the problem.

Reply:
Metadata is always `most significant = last`. so whatever is putting them in the wrong order is what needs the fix.

Reply:
`SelectorModel selector` is the one from `ActionModel` `IList<object> metadata` is the controller metadata. We're combining the two. We used to append `metadata` to the `selector.EndpointMetadata`. This would mean you got `SelectorModel.EndpointMetadata = { ActionMetadata, ControllerMetadata }` prior to this change at the end of this call. This change makes it so that you get `SelectorModel.EndpointMetadata = {  ControllerMetadata, ActionMetadata }`

see https://github.com/aspnet/AspNetCore/pull/9099/files#diff-e47afec1f608e7785f1bd3b4e4c6427eL277 for an example of this

Reply:
Reversing a list is not the correct solution. It means that some part of the system is using the wrong representation and possibly exposing it to extensibility.